src/Illuminate/Database/Eloquent/FactoryBuilder.php

```diff
@@ -92,12 +92,12 @@ public function times($amount)
     /**
      * Set the states to be applied to the model.
      *
-     * @param  array|mixed  $states
+     * @param  array  ...$states
      * @return $this
      */
-    public function states($states)
+    public function states(...$states)
     {
-        $this->activeStates = is_array($states) ? $states : func_get_args();
+        $this->activeStates = $states;
```

Comment:
Previous behaviour `states(['foo', 'bar'])` is no longer accepted here, might worth considering that.